src/QirRuntime/test/QIR-tracer/tracer-target.qs

```diff
@@ -0,0 +1,297 @@
+    }
+
+    @TargetInstruction("single_qubit_measure")
+    operation single_qubit_measure(op_id: Int, duration: Int, qb : Qubit) : Result {
+        body intrinsic;
+    }
+
+    @TargetInstruction("joint_measure")
+    operation multi_qubit_measure(op_id: Int, duration: Int, qbs : Qubit[]) : Result {
+        body intrinsic;
+    }
+}
+
+
+namespace Microsoft.Quantum.Intrinsic {
+
+    open Microsoft.Quantum.Core;
+    open Microsoft.Quantum.Instructions as Phys;
+    open Microsoft.Quantum.Targeting;
+
+    @Inline()
+    operation X(qb : Qubit) : Unit
+    is Adj + Ctl {
+        body  (...) { Phys.single_qubit_op(0, 1, qb); }
+        adjoint (...) { Phys.single_qubit_op(0, 1, qb); }
+        controlled (ctls, ...)
+        {
+            if (Length(ctls) == 1) { Phys.single_qubit_op_ctl(1, 1, ctls, qb); }
+            else { Phys.single_qubit_op_ctl(2, 1, ctls, qb); }
+        }
+        controlled adjoint (ctls, ...)
+        {
+            if (Length(ctls) == 1) { Phys.single_qubit_op_ctl(1, 1, ctls, qb); }
+            else { Phys.single_qubit_op_ctl(2, 1, ctls, qb); }
+        }
+    }
+
+    operation CNOT(control : Qubit, target : Qubit) : Unit
+    is Adj {
+        body  (...) { Controlled X([control], target); }
+        adjoint (...) { Controlled X([control], target); }
+    }
+
+    @Inline()
+    operation Y(qb : Qubit) : Unit
+    is Adj + Ctl{
+        body  (...) { Phys.single_qubit_op(3, 1, qb); }
+        adjoint (...) { Phys.single_qubit_op(3, 1, qb); }
+        controlled (ctls, ...)
+        {
+            if (Length(ctls) == 1) { Phys.single_qubit_op_ctl(4, 1, ctls, qb); }
+            else { Phys.single_qubit_op_ctl(5, 1, ctls, qb); }
+        }
+        controlled adjoint (ctls, ...)
+        {
+            if (Length(ctls) == 1) { Phys.single_qubit_op_ctl(4, 1, ctls, qb); }
+            else { Phys.single_qubit_op_ctl(5, 1, ctls, qb); }
+        }
+    }
+
+    @Inline()
+    operation Z(qb : Qubit) : Unit
+    is Adj + Ctl {
+        body  (...) { Phys.single_qubit_op(6, 1, qb); }
+        adjoint (...) { Phys.single_qubit_op(6, 1, qb); }
+        controlled (ctls, ...)
+        {
+            if (Length(ctls) == 1) { Phys.single_qubit_op_ctl(7, 1, ctls, qb); }
+            else { Phys.single_qubit_op_ctl(8, 1, ctls, qb); }
+        }
+        controlled adjoint (ctls, ...)
+        {
+            if (Length(ctls) == 1) { Phys.single_qubit_op_ctl(7, 1, ctls, qb); }
+            else { Phys.single_qubit_op_ctl(8, 1, ctls, qb); }
+        }
+    }
+
+    @Inline()
+    operation H(qb : Qubit) : Unit
+    is Adj + Ctl {
+        body  (...) { Phys.single_qubit_op(9, 1, qb); }
+        adjoint (...) { Phys.single_qubit_op(9, 1, qb); }
+        controlled (ctls, ...)
+        {
+            Phys.single_qubit_op_ctl(10, 1, ctls, qb);
+        }
+        controlled adjoint (ctls, ...)
+        {
+            Phys.single_qubit_op_ctl(10, 1, ctls, qb);
+        }
+    }
+
+    @Inline()
+    operation Tz(qb : Qubit) : Unit
```

Comment:
This is defined in the Microsoft.Quantum.Intrinsic namespace, but doesn't match any existing intrinsic operations that are already a part of the Q# standard library; is this operation exposed anywhere publicly?

Reply:
These are needed for Hadamard tracking so moved into `Microsoft.Quantum.Instructions` namespace